py/src/braintrust/logger.py

```diff
@@ -1852,11 +1877,15 @@ def __init__(
             _EXEC_COUNTER += 1
             exec_counter = _EXEC_COUNTER
 
+        type_arg = dict()
+        if type is not None:
+            type_arg = dict(type=type)
         self.internal_data = dict(
             metrics=dict(
                 start=start_time or time.time(),
             ),
-            span_attributes=dict(**span_attributes, name=name, exec_counter=exec_counter),
+            # Set type first, in case they override it in `span_attributes`.
```

Comment:
Should we do that for name as well? Technically a breaking change, but it feels pretty low-likelihood that someone specifies a name inside `span_attributes` instead of the hoisted name param and expects it to get overriden by the default name.

Reply:
Yeah good idea